src/renderer/base/thread.cpp

```diff
@@ -162,17 +162,17 @@ DWORD WINAPI RenderThread::_ThreadProc()
     {
         WaitForSingleObject(_hPaintEnabledEvent, INFINITE);
 
-        if (!_fNextFrameRequested.exchange(false))
+        if (!_fNextFrameRequested.exchange(false, std::memory_order_acq_rel))
         {
             // <--
             // If `NotifyPaint` is called at this point, then it will not
             // set the event because `_fWaiting` is not `true` yet so we have
             // to check again below.
 
-            _fWaiting.store(true);
+            _fWaiting.store(true, std::memory_order_release);

```

Comment:
So if I understand it properly, this one probably _shouldn't_ be relaxed because you're checking two different atomics instead of just one (and hoping for consistency across both), right?

Reply:
It can be "relaxed" because the "release" would make prior writes from this thread visible to other threads who "acquire" `_fWaiting`'s value. But in this case the only one "acquiring" its value is `NotifyPaint()`, which doesn't need any prior writes done by `_ThreadProc`, since the only thing `NotifyPaint()` does is write to `_fNextFrameRequested` (but it doesn't rely upon its value, nor does it rely on any other value).
(This information is supplied without liability. ⚖😄)